### PR TITLE
Add prrte debug examples and test framework to CI tests

### PR DESCRIPTION
**Pull request by drwootton**
Add scripts to build and run prrte/examples/debugger examples to CI test suite, using source from that directory.
Add tc  filter.c and test-utils.c as support code for these tests
See README.md in the prrte/debug directory for this pull request for detailed explanation.

### CODE REVIEW

**Comment by jjhursey** [comment]
For the `python36` issue can we change it to `python3` (the build box has `python3     python3.6   python3.6m`)

**Comment by jjhursey** [comment]
Can you squash the commits down to 1 or 2?

Something like:
```
git rebase HEAD~12
# Then change 'pick' to 'squash' on every line but the first. It'll let you re-edit the commit message.
git push -f
```

**Comment by rhc54** [comment]
> For the `python36` issue can we change it to `python3` (the build box has `python3 python3.6 python3.6m`)

Do we need to tie this into the configure logic's detection of Python? Or do we just leave it available and assume the person running the tests has an appropriate Python? Not advocating either way - just asking.

**Comment by drwootton** [comment]
If we need to hook this into configure, I will need to work with @jjhursey since I haven't dealt much with configure scripts. The run.py script does require some version of Python 3, so allowing configure to pick an arbitrary Python would be a problem. I just tried Python 2.7 and the script immediately fails at an import for DEVNULL.

I don't really like hard coding python3.6 either. This seems rather fragile if there's not an explicit commitment to keeping this version in any OS distributions we run on.

**Comment by jjhursey** [comment]
I have an idea for how to handle the `python3.6` discovery. Let me tinker with it a bit and I'll post back.

**Comment by jjhursey** [comment]
bot:retest

File: prrte/debug/run.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+# Wrapper script used by CI framework to invoke test cases in this directory
+./run.py
```
Comment (jjhursey) [inline prrte/debug/run.sh:3]: To address the python binary versioning replace the `run.sh` script with the following:

```bash
#!/bin/bash

_python=""

# Find all python* binaries
_CHECK_BINARIES=()
_CHECK_BINARIES+=`compgen -c python | grep -v "config\|m$" | sort -ru`

for _py in ${_CHECK_BINARIES} ; do
    _py_fp=`command -v ${_py}`
    # Check if we have this binary installed
    if [ $? -eq 0 ] ; then
        # Extract the major version
        VER=`${_py_fp} -c 'import sys; print(sys.version_info[0])'`
        # Find the first one at level 3 or above
        if [ $VER -ge 3 ] ; then
            _python=${_py_fp}
            break
        fi
    fi
done

# Wrapper script used by CI framework to invoke test cases in this directory
${_python} ./run.py
exit $?
```

**Comment by jjhursey** [comment]
**Re: python binary version**
I just added a bash script that will detect the python binary name that is at version 3 or above. If none is found then it exits in error. Since CI runs the `run.sh` script then this should be enough to find the correct interpreter.

**Review by jjhursey — APPROVED** [review]

**Comment by jjhursey** [comment]
There is still more work to do to enable more debugger test cases. So we will add more to this as things improve on master.

**Pull request merged**